packages/presence-signal/src/createPresenceSignal.ts

```diff
@@ -0,0 +1,108 @@
+import { createSignal, createEffect, onCleanup, type Accessor } from "solid-js";
+
+type SharedTransitionConfig = {
+  /** Duration in milliseconds used both for enter and exit transitions. */
+  transitionDuration: number;
+};
+
+type SeparateTransitionConfig = {
+  /** Duration in milliseconds used for enter transitions (overrides `transitionDuration` if provided). */
+  enterTransitionDuration: number;
+  /** Duration in milliseconds used for exit transitions (overrides `transitionDuration` if provided). */
+  exitTransitionDuration: number;
+};
+
+type Options = (
+  | SharedTransitionConfig
+  | SeparateTransitionConfig
+  | (SharedTransitionConfig & SeparateTransitionConfig)
+) & {
+  /** Opt-in to animating the entering of an element if `isVisible` is `true` during the initial mount. */
+  initialEnter?: boolean;
+};
+
+/**
+ * Animates the appearance of its children.
+ */
+export default function createPresenceSignal(
+  /** Indicates whether the component that the resulting values will be used upon should be visible to the user. */
+  isVisible: Accessor<boolean>,
```

Comment:
why not just call it `source` or `sourceState`? The source signal having the same name as the returned `isVisible` makes it a bit confusing, you might think that this state is getting returned directly when it's not.

Reply:
I agree that the overloaded language is not optional. How would you feel about `preferredState` ❓

Reply:
imo keeping it as `source` or `input` make sense as this is a reactive source that drives this primitive. Much like the `source` param in `createResource` / `createMemo`
https://github.com/solidjs/solid/blob/4dd99a7a590b5baa32175198c6c2013d6584c6fb/packages/solid/src/reactive/signal.ts#L569

Reply:
I'll go with `source` for the moment on this one. Though per our discussion in a future commit, I'm just going to merge them anyways, in which case this argument will just become `item`.